### PR TITLE
Fix back button

### LINKED ISSUE #233: _bindPopChangeHandler event.state is null on initial load of category page
Issue Brief
--
Sometimes when using the browser back button, the product grid in the category doesn't get updated.

After debugging we noticed on initial load of a category, there is no ajax call being done yet, therefore the event.state is not yet updated through the 'this._updateState(response);' function in navigation-form.js:198.
So when going back using the browser button event.state is empty in '_bindPopChangeHandler: function(){' causing the content not to be updated.

A possible workaround/fix we found is adding this to the _hookEvents function within the `if (this.options.ajaxFilters) {`  in navigation-form.js:30

```  
if (this.options.ajaxFilters) {
    if (!$('.filter-actions').length) {
        $.ajax({
            url: this.options.ajaxEndpoint,
            data: this._getFilterParameters(),
            cache: true,
            success: function (response) {
                this._updateState(response);
            }.bind(this),
            error: function (jqXHR, errorStatus) {
                if (errorStatus !== 'abort') {
                    // Something went wrong, try to navigate to the selected filter
                    this._defaultHandler(event);
                }
            }.bind(this)
        });
    }
    
    this._bindPopChangeHandler();
}
```                        

The `if (!$('.filter-actions').length) {` statement is just a test, it's probably better to set a flag to determine if event.state has been set at least once. So the idea is that `this._updateState(response);` should be triggered at least once to make sure there is always a state to go back to. Hopefully it will give at least some idea on how/where to fix it.

Environment
--
* PHP Version: 7.4
* Magento Version: 2.4.3-p1
* Tweakwise Version: 4.1.0
* Tweakwise Export Version: 4.0.1
* Magento Deploy Mode: production,development

Steps to reproduce
--
1. [Go to a category](https://tweakwise2-ee.emico.nl/men/tops-men/hoodies-and-sweatshirts-men.html) (having ajax enabled) 
2. Apply just one filter, do not trigger another ajax call (navigation or an additional filter)
3. Go back using the browser back button

Actual result
--
1. Product grid in the category is not updated with the 'original' state, instead it still shows the 'filtered' result

Expected result
--
- [ ] Product grid in a category shows the original content which reflects the state of the filters ( in this case no filters applied)

### LINKED ISSUE #239: Browse back button on POP doesn't work (navigation issue)
Issue Brief
--
The "back" button of my browser (Chrome on MacOS) doesn't reload the products on a POP. 

Case is when we're on a POP and navigate to the second page. The products are loaded using AJAX and the URL changes to `http://example.com/pop/?p=2`. This is correct. However, if we now click the back button of our browser, the URL changes back to what it was, but nothing else happens. The products are not reloaded again and the navigation buttons still show that the second page is active.

See a case at the following video:  https://watch.screencastify.com/v/ZKWfv88YVf9mObEc3XCF 

Environment
--
* PHP Version: 7.4
* Magento Version: 2.4.2-p1
* Tweakwise Version: 4.1.2
* Tweakwise Export Version: 4.0.3
* Magento Deploy Mode: production

Steps to reproduce
--
1. See video:  https://watch.screencastify.com/v/ZKWfv88YVf9mObEc3XCF

Actual result
--
1. Products are still the products from page 2.

Expected result
--
List the expected results as a bullet list of expectations
- Products are reloaded according to history.

### PR DESCRIPTION
**Pull request by ahuininga-orisha**
Fixes #239 and also fixes #233 (seems to be the same issue)

### CODE REVIEW

**Comment by mischabraam** [comment]
Applied this change as patch with no result.

**Comment by mischabraam** [comment]
Applied this change as patch with no result.

**Comment by ahuininga-orisha** [comment]
> Applied this change as patch with no result.

Did you clear the cache (magento cache and browser cache) and redeploy the static content? 

Which version of the tweak wise module do you use? The code only works in the latest version. If you don't have the latest version you need to remove the following from the if on line 45
this.options.ajaxCache

**Comment by mischabraam** [comment]
Cool, I wasn't on the latest version. Updated to the latest version, applied this patch and it works. Thanks!

File: view/frontend/web/js/navigation-form.js [hunk 1]
```diff
@@ -32,9 +32,30 @@ define([
...
             return this._superApply(arguments);
         },
 
+        /**
+         * Fix first page history on page load when using ajax filters
+         *
+         *  @private
+         */
+        _fixAjaxHistory: function () {
+            if(this.options.ajaxFilters && this.options.ajaxCache && ((!window.history.state) || (!window.history.state.html)))
```
Comment (bramstroker) [inline view/frontend/web/js/navigation-form.js]: Remove superfluous parentheses

```suggestion
            if(this.options.ajaxFilters && this.options.ajaxCache && (!window.history.state || !window.history.state.html))
```

**Pull request merged**